src/metabase/util/malli/schema.clj

```diff
@@ -55,7 +56,7 @@
 (def NonBlankString
   "Schema for a string that cannot be blank."
   (mu/with-api-error-message
-    [:string {:min 1}]
+    [:fn (fn [s] (and (string? s) (not (str/blank? s))))]
```

Comment:
We actually have a version of this in `metabase.lib.schema.common` defined like

```clj
(mr/def ::non-blank-string
  [:and
   [:string {:min 1}]
   [:fn
    {:error/message "non-blank string"}
    (complement str/blank?)]])
```

Maybe it would be best just to change this to use that as a starting point e.g.

```clj
(mu/with-api-error-message ::lib.schema.common/non-blank-string (deferred-tru ...))
```

For `:fn` schemas you should always include an `:error/message` or `:error/fn` because otherwise Malli will just report failures as `"Unknown error"` which is hard to debug.

Reply:
nice, changed it in https://github.com/metabase/metabase/pull/32070/commits/c4c09bfae784de0da3048b101100e2c46870e2a1